Mail/Views/New Message/AttachmentsManager.swift

```diff
@@ -0,0 +1,266 @@
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    func addCameraAttachment(
+        data: Data,
+        disposition: AttachmentDisposition = .attachment,
+        completion: @escaping (String) -> Void = { _ in
+            // TODO: - Manage inline attachment
+        }
+    ) {
+        Task {
+            let temporaryURL = FileManager.default.temporaryDirectory.appendingPathComponent(UUID().uuidString)
+            let temporaryFileURL = temporaryURL.appendingPathComponent(getDefaultFileName()).appendingPathExtension("jpeg")
+            let localAttachment = await self.createLocalAttachment(name: temporaryFileURL.lastPathComponent,
+                                                                   type: UTType.image,
+                                                                   disposition: .attachment)
+            do {
+                try FileManager.default.createDirectory(at: temporaryURL, withIntermediateDirectories: true)
+                try data.write(to: temporaryFileURL)
+
+                let updatedAttachment = await self.updateLocalAttachment(url: temporaryURL, attachment: localAttachment)
+                let remoteAttachment = try await self.sendAttachment(url: temporaryFileURL, localAttachment: updatedAttachment)
+
+                if disposition == .inline, let cid = remoteAttachment?.contentId {
+                    completion(cid)
+                }
+            } catch {
+                DDLogError("Error while creating attachment: \(error.localizedDescription)")
+                await updateAttachmentUploadError(attachment: localAttachment, error: error)
+            }
+        }
+    }
+
+    func loadFileRepresentation(_ itemProvider: NSItemProvider, typeIdentifier: String) async throws -> URL {
+        try await withCheckedThrowingContinuation { continuation in
+            itemProvider.loadFileRepresentation(forTypeIdentifier: typeIdentifier) { fileProviderURL, error in
+                if let fileProviderURL {
+                    do {
+                        let temporaryURL = FileManager.default.temporaryDirectory.appendingPathComponent(UUID().uuidString)
+                        let temporaryFileURL = temporaryURL.appendingPathComponent(fileProviderURL.lastPathComponent)
+                        try FileManager.default.createDirectory(at: temporaryURL, withIntermediateDirectories: true)
+                        try FileManager.default.copyItem(atPath: fileProviderURL.path, toPath: temporaryFileURL.path)
+                        continuation.resume(returning: temporaryFileURL)
+                    } catch {
+                        continuation.resume(throwing: error)
+                    }
+                } else {
+                    continuation.resume(throwing: error ?? MailError.unknownError)
+                }
+            }
+        }
+    }
+
+    private func getDefaultFileName() -> String {
```

Comment:
Maybe this is an old trick no longer needed, but lazy loading and reusing the date formatter configured with a format was a thing.